src/steps/secret-manager/index.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  createDirectRelationship,
+  getRawData,
+  IntegrationStep,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+import { secretmanager_v1beta1 } from 'googleapis';
+import { IntegrationConfig, IntegrationStepContext } from '../../types';
+import { SecretManagerClient } from './client';
+import {
+  ENTITY_CLASS_SECRET_MANAGER_SECRET,
+  ENTITY_CLASS_SECRET_MANAGER_SECRET_VERSION,
+  ENTITY_TYPE_SECRET_MANAGER_SECRET,
+  ENTITY_TYPE_SECRET_MANAGER_SECRET_VERSION,
+  RELATIONSHIP_TYPE_SECRET_HAS_VERSION,
+  STEP_SECRET_MANAGER_FETCH_SECRETS,
+  STEP_SECRET_MANAGER_FETCH_SECRET_VERSION,
+} from './constants';
+import { createSecretEntity, createSecretVersionEntity } from './converters';
+
+export async function fetchSecrets(
+  context: IntegrationStepContext,
+): Promise<void> {
+  const {
+    jobState,
+    instance: { config },
+  } = context;
+  const client = new SecretManagerClient({ config });
+
+  await client.iterateSecrets(async (secret) => {
+    await jobState.addEntity(createSecretEntity(secret));
+  });
+}
+
+export async function fetchSecretVersions(
+  context: IntegrationStepContext,
+): Promise<void> {
+  const {
+    jobState,
+    instance: { config },
+  } = context;
+  const client = new SecretManagerClient({ config });
+
+  await jobState.iterateEntities(
+    { _type: ENTITY_TYPE_SECRET_MANAGER_SECRET },
+    async (secretEntity) => {
+      const secret =
+        getRawData<secretmanager_v1beta1.Schema$Secret>(secretEntity);
+
+      // TO-DO: Why is this necessary?
+      if (!secret) return;
+
+      await client.iterateSecretVersions(secret, async (version) => {
```

Comment:
Per your comment above, you can instead assert that `secret` is present:


```suggestion
      await client.iterateSecretVersions(secret!, async (version) => {
```